SplitFiles/bread.c

```diff
@@ -39,4 +67,105 @@ BreadBin* get_bread_bin() {
     bin->probTotal = 0;
     bin->breadTypes = types;
 
-}
\ No newline at end of file
+    /*******************************************************
+    ***** This is where we establish the Bread types
+    ***** Once established they are only ever referenced
+    ***** through the BreadBin methods
+    ********************************************************/
+    add_bread(bin, 
+        create_bread_type(
+            'W', //letter
+            "White", //desc
+            62, //thermal Mass
+            1, //variability
+            64, //moisture
+            48, // moisture variability
+            50, // cost
+            64  //callProb
+        )
+    );
+
+    add_bread(bin, 
+        create_bread_type(
+            'B', //letter
+            "Brown", //desc
+            82, //thermal Mass
+            10, //variability
+            96, //moisture
+            64, // moisture variability
+            75, // cost
+            32  //callProb
+        )
+    );
+
+    add_bread(bin, 
+        create_bread_type(
+            'G', //letter
+            "baGel", //desc
+            150, //thermal Mass
+            28, //variability
+            64, //moisture
+            192, // moisture variability
+            125, // cost
+            16  //callProb
+        )
+    );
+
+    add_bread(bin, 
+        create_bread_type(
+            'C', //letter
+            "Ciabatta", //desc
+            180, //thermal Mass
+            40, //variability
+            32, //moisture
+            12, // moisture variability
+            100, // cost
+            12  //callProb
+        )
+   );
+   return bin;
+}
+
+
+BreadType*  create_bread_type(
+    unsigned char     letter, 
+    char*             desc,
+    int               thermalMass,
+    int               massVariability,  //Plus or minus this amount
+    int               moisture,         //This moisture
+    int               moistureVariability,  //Plus or minus...
+    int               cost,
+    int               callProb         //Actually an integer, higher number = more likely
+) {
+        BreadType* newType = malloc(sizeof(BreadType));
+        newType->letter = letter;
+        newType->thermalMass = thermalMass;
+        newType->massVariability = massVariability;
+        newType->moisture = moisture;
+        newType->moistureVariability = moistureVariability;
+        newType->cost = cost;
+        newType->callProb = callProb;
+        newType->desc = desc;
+        return newType;
+    }
+
+int main() {
+    start_frame_count();
+    //Set up the bin
+    BreadBin* bin = get_bread_bin();
+    BreadType* type;
+    BreadState* bread;
+    int startFrame, endFrame;
+    while (1) {
+        startFrame = G_frames;
+        endFrame = startFrame + 50;
+        type = rand_bread_type(bin);
+        printf(PRINTAT"\x05\x05""%s      ", type->desc);
+        bread = get_bread(bin, type->letter);
```

Comment:
This call to be used when the player wants to toast a slice.  If bread == NULL then the bread type is unrecognised.

We may want to add a "get_type" method to allow a pre-check of a command/keystroke